src/pages/GamePage.jsx

```diff
@@ -0,0 +1,117 @@
+import React, {useState} from 'react';
+import Menu from '../Components/SidebarMenu/Menu';
+import { Container, Row, Col } from 'react-bootstrap';
+
+const numbersQuantity = 20;
+const numbersRange = 20; 
+const ranArr = Array.from({length:numbersQuantity},() => Math.floor(Math.random() * numbersRange));
+
+const GamePage = () => {
+    const [userNumber, setUserNumber] = useState(false), 
+          [userNumberHandler, setUserNumberHandler] = useState(false),
+          [errorHandler, setErrorHandler]= useState(false),
+          [resultHandler, setResultHandler]= useState(false);
+   
+    const inputHandler = (text) => {
+      setUserNumberHandler(parseInt(text.target.value))
+    };
+    const saveUserNumber = (event) => {
+      if (isNaN(userNumberHandler)) {
+        setErrorHandler('Podaj cyfre')
+      } else {
+      event.preventDefault();
+      setUserNumber(userNumberHandler);
+      setErrorHandler(false);
+      iterArray(ranArr);
+      }
+    };
+    const keyHolder = (event) =>{
+      if (event.key === 'enter' && typeof userNumberHandler === 'number') {
+        setUserNumber(userNumberHandler);
+      }
+    };
+    const iterArray = (array) => {
+      let i = 0;
+      setResultHandler(false)
+      do {
+        if (array[i]===userNumber){
+          setResultHandler(true)
+          console.log(`traf na iteracji ${i} wynik ${resultHandler}`);
+          i+=1;
+        } else {
+          i+=1;
```

Comment:
podwajasz tą inkrementację. Skoro jest w jednym i drugim warunku, to powinna być raz poza blokiem  `if else`